### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by Anonoei**
Fixed typos in comments
Fixed typo in declaration in wrappers/themis/java...SecureSession.java:
  Renamed jniGenerateConntect to jniGenerateConnect

## Checklist
- [x] Change is covered by automated tests
- [x] The [coding guidelines] are followed
- [x] Public API has proper documentation
- [x] Example projects and code samples are up-to-date (in case of API changes)
- [x] ~~Changelog is updated~~ (not needed, despite JNI changes)

[coding guidelines]: https://github.com/cossacklabs/themis/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by vixentael** [comment]
Thank you @Anonoei! Some of these are just British spelling (tokenisation, centralised, etc), but many are just typos. I'll review.

I'm curious why have you decided to fix typos?

**Comment by Anonoei** [comment]
I didn't know that!
I noticed some British spelling (like "favourite"), but didn't know the rest were.

Why:
I'm writing an end-to-end encrypted messenger, and decided to use Themis for it and while creating abstractions for functions and classes I noticed typos and decided I'd like to contribute to the project. (I also use VSCode and have an extension which highlights typos)

File: README.md
Comment (ilammy) [inline README.md:141]: Revert to British: analyse
```suggestion
We use a lot of automated security testing, i.e. static code analysers, fuzzing tools, memory analysers, unit tests (per each platform), integration tests (to find compatibility issues between different Themis-supported languages, OS and x86/x64 architectures). Read more about our security testing practices in [Themis security docs](https://docs.cossacklabs.com/themis/security/).
```

File: include/soter/soter_kdf.h
Comment (ilammy) [inline include/soter/soter_kdf.h:142]: Good: Acra docs use this spelling.

File: src/soter/openssl/soter_asym_ka.c
Comment (ilammy) [inline src/soter/openssl/soter_asym_ka.c:171]: Okay. This has been a historical point of contention, now I think is a good time to make a decision.

I propose to bless “key pair” as the preferred spelling in prose, and `KeyPair` or `key_pair` in code.

Rationale: this spelling is used in JP's *Serious Cryptography*.

File: src/wrappers/themis/java/com/cossacklabs/themis/SecureSession.java
Comment (ilammy) [inline src/wrappers/themis/java/com/cossacklabs/themis/SecureSession.java:119]: I support fixing the function name. However, this needs a bit more work.

---

First, the function name needs to be updated in C code as well:

https://github.com/cossacklabs/themis/blob/5d4a0b707c33b32d7a1f2afc0805fb7b1fa49b8d/jni/themis_session.c#L469-L470

```diff
-Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect
+Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect
```

(mind the code format, it will require parameter list to be neatly aligned)

---

Second, even though JNI interface is a private one, I'd keep the old symbol name around to avoid breaking applications that were built using the old JavaThemis. When they upgrade the JNI library, they should keep working without having to recompile their Java code using the new version of JavaThemis. 

```c
JNIEXPORT jbyteArray JNICALL Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect(JNIEnv* env, 
                                                                                           jobject thiz)
__attribute__((alias("Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect")));
```

With this added, users of the new version will use the "good" name, and any new code should be written using this name. However, the function will also be available under the old name for binaries that were linked against that name.

File: include/soter/soter_asym_cipher.h
Comment (ilammy) [inline include/soter/soter_asym_cipher.h:72]: If we go around fixing typos...
```suggestion
 * @note If cipher_data==NULL or cipher_data_length less than needed to store cipher data, @ref
```
(the same “then need” mistake is made in a bunch of other places)

**Review by ilammy — CHANGES_REQUESTED** [review]
Looks good overall, but the JNI function rename needs to a bit more work.

Thank you, @Anonoei!

**Comment by Anonoei** [comment]
@ilammy, let me know if any changes need to be reverted or if you have more suggestions. Thanks for the informative feedback and providing reasons for your decisions!

**Review by ilammy — APPROVED** [review]
All good 👍

**Comment by ilammy** [comment]
@Anonoei,

It seems `clang-format` wants the attribute to be indented:

<details>
<summary>show diff</summary>

```diff
--- jni/themis_session.c	2022-10-31 12:23:29.871825157 +0000
+++ -	2022-10-31 12:24:41.699042533 +0000
@@ -527,9 +527,9 @@
 
     return output;
 }
-JNIEXPORT jbyteArray JNICALL Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect(JNIEnv* env, 
+JNIEXPORT jbyteArray JNICALL Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect(JNIEnv* env,
                                                                                            jobject thiz)
-__attribute__((alias("Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect")));
+    __attribute__((alias("Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect")));
 
 JNIEXPORT jlong JNICALL Java_com_cossacklabs_themis_SecureSession_create(JNIEnv* env,
                                                                          jobject thiz,
```

</details>

Sorry for not remembering that in my initial suggestion.

---

Although, it also seems that aliases are not supported on macOS:

```
make: *** [build/obj/jni/themis_session.c.o] Error 1
cc -c -o build/obj/jni/themis_session.c.o jni/themis_session.c -Iinclude -Isrc -Isrc/wrappers/themis/ -I/usr/local/include -fPIC -DLIBRESSL -DCRYPTO_ENGINE_PATH=openssl -I/usr/local/opt/openssl@1.1/include -O2 -g -fno-omit-frame-pointer -fstack-protector -D_FORTIFY_SOURCE=2 -Werror -Wall -Wextra -Wformat -Wformat-nonliteral -Wformat-security -Wstrict-overflow -Wswitch -Wvla -Wpointer-arith -fvisibility=hidden  -I/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.345-1/x64/Contents/Home//include -I/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.345-1/x64/Contents/Home//include/darwin
jni/themis_session.c:532:16: error: aliases are not supported on darwin
__attribute__((alias("Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect")));
               ^
1 error generated.
```

So it's probably better just define `Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect()` to call `Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect()`. I don't think it will affect performance all too much.

---

There are other build failures around Java code. I'm not sure for how long those are broken, but they don't seem to be either related to the recent changes, or some transient flukes.

@vixentael, are willing to disregard those other issues?

**Comment by Anonoei** [comment]
@ilammy 

I removed alias in favor of a function call as requested.

File: jni/themis_session.c
Comment (ilammy) [inline jni/themis_session.c:531]: Looks like formatting checks are still failing 😞

<details>
<summary>CI failure</summary>

```
check jni/themis_session.c 12G[ERRORS]
clang-format-8 jni/themis_session.c | diff -u jni/themis_session.c - && clang-tidy-8 jni/themis_session.c -- -Iinclude -Isrc -Isrc/wrappers/themis/ -I/usr/local/include -fPIC -DBORINGSSL -DCRYPTO_ENGINE_PATH=boringssl -Ithird_party/boringssl/src/include -O2 -g -fno-omit-frame-pointer -fstack-protector -D_FORTIFY_SOURCE=2 -Werror -Wall -Wextra -Wformat -Wformat-nonliteral -Wformat-security -Wnull-dereference -Wshift-overflow -Wshift-negative-value -Wstrict-overflow -Wswitch -Wvla -Wpointer-arith -Wstrict-prototypes -fvisibility=hidden  -I/usr/lib/jvm/java-11-openjdk-amd64/include -I/usr/lib/jvm/java-11-openjdk-amd64/include/linux 2>/dev/null && touch build/obj/jni/themis_session.c.fmt_check
--- jni/themis_session.c	2022-11-01 12:28:16.036063376 +0000
+++ -	2022-11-01 12:29:12.077085163 +0000
@@ -527,7 +527,7 @@
 
     return output;
 }
-JNIEXPORT jbyteArray JNICALL Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect(JNIEnv* env, 
+JNIEXPORT jbyteArray JNICALL Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect(JNIEnv* env,
                                                                                            jobject thiz)
 {
     return Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect(env, thiz);
make: *** [Makefile:452: build/obj/jni/themis_session.c.fmt_check] Error 1
```

</details>

It seems they are triggered by this stray whitespace which is not immediately obvious in the diff:

```suggestion
JNIEXPORT jbyteArray JNICALL Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect(JNIEnv* env,
                                                                                           jobject thiz)
```

File: src/wrappers/themis/rust/src/lib.rs
Comment (vixentael) [inline src/wrappers/themis/rust/src/lib.rs:20]: that's just british spelling here
https://www.merriam-webster.com/words-at-play/traveling-vs-travelling-usage

```suggestion
//! in your application) and data in motion (i.e. travelling between client and server, server
```

**Review by vixentael — CHANGES_REQUESTED** [review]
Thank you @Anonoei! 

I reviewed all the changes, this is a good PR!

Let's fix a tricky whitespace and reverse one british spelling, and we should good to merge.

**Comment by Anonoei** [comment]
@vixentael 
Suggestions committed!

**Review by ilammy — APPROVED** [review]
Alrighty... One final thing left to resolve: merge conflicts.

@Anonoei, could you please rebase your changes onto the latest `master`?

**Comment by Anonoei** [comment]
@ilammy It should be up to date with themis/master now.

**Comment by vixentael** [comment]
@Anonoei  i have a feeling that you've merged the changes from cossacklabs/master to Anonoei/master instead of rebasing 🤔

**Comment by ilammy** [comment]
> i have a feeling that you've merged the changes from cossacklabs/master to Anonoei/master instead of rebasing 

GitHub usually handles that decently as well, but for some reason this time GitHub decided to derp harder than usual.

"Non-conforming" PRs have very poor support in GitHub UI, unfortunately 😞 And that's not changing anytime soon. That's why I suggested `rebase` because reviewing merge commits with conflict resolution is literally impossible on GitHub, and it's quite arcane with git in general.

In this case the update branch overwrites the updates applied to `docs/examples/react-native/react-native-example/App.js` on master, which is not nice.

This is the diff that should be applied: https://github.com/cossacklabs/themis/compare/cossacklabs:71b3aaa...Anonoei:a241572#diff-cd5d1fc8e25d9138f705c824a91941c51894000aec5c00492975136bf3df0711 (along with all other changes).

I don't think @Anonoei has to make any other changes, so I'll take the liberty to do the rebase...

**Review by vixentael — APPROVED** [review]

**Comment by vixentael** [comment]
Let's wait for tests and merge then?

**Comment by ilammy** [comment]
@Anonoei, I'm sorry for being intrusive and force-pushing into your repo.

Since you're using `master` for your changes, once the PR is merged you will need to reset your branch to the `cossacklabs/themis:master` if you want to track our repo and maybe contribute more in the future 🥺

```bash
git fetch upstream
git checkout master
git reset --hard upstream/master
```

(**Don't** do it right now. Only after PR is merged.)

I assume "origin" is your fork and "upstream" is our repo:

```console
$ git remote -v
origin		git@github.com:Anonoei/themis.git (fetch)
origin		git@github.com:Anonoei/themis.git (push)
upstream	git@github.com:cossacklabs/themis.git (fetch)
upstream	git@github.com:cossacklabs/themis.git (push)
```

If not, you can add it like this before reciting the spells above:

```bash
git remote add upstream git@github.com:cossacklabs/themis.git
```

**Comment by ilammy** [comment]
Okay... I restarted the build of Objective-C examples several times and it keeps failing the Carthage examples. Something-something about “Package.resolved file is corrupted or malformed”, so it's probably not a fluke. @vixentael, @julepka, you might want to take a look at it sometime.

I don't think those failures are related to this changeset, so I've merged it as is.

Thank you for your contribution, @Anonoei! ❤️

**Comment by vixentael** [comment]
@ilammy these are the same Carthage issues that we are facing for two years — Carthage tries to build every project it finds. It wasn't solved on Carthage side (and I guess never will be), and sometimes our workarounds stop working and need more workarounding. So, expected.

---

Thank you for your contribution @Anonoei ❤️
Thank you for handling this large PR @ilammy ❤️🖤

**Comment by Anonoei** [comment]
@vixentael @ilammy 
Hey all, sorry for the late reply
Sorry I messed up the rebase, and thank you for all the help with the PR.
I really appreciate everything you have done and I'd love to contribute in the future :)
Thanks again!

**Pull request merged**
